src/dc_tools.rs

```diff
@@ -2132,4 +2116,28 @@ mod tests {
             assert_eq!(cmp, 0);
         }
     }
+
+    #[test]
+    fn test_dc_extract_grpid_from_rfc724_mid() {
+        // Should return 0 if we pass invalid mid
+        let str = "foobar";
```

Comment:
I'm surprised using str as a variable is even possible, I thought this was a keyword.  Is it wise though?

Reply:
Maybe not really wise but probably fine from a test perspective. But calling the variable mid would be better.